cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/AppFabricServer.java

```diff
@@ -176,6 +184,27 @@ protected void startUp() throws Exception {
                                           AppMetadataStore.create(context).getApplicationCount());
     metricsCollectionService.getContext(Collections.emptyMap()).gauge(Constants.Metrics.Program.APPLICATION_COUNT,
                                                                       count);
+
+    Map<ImmutablePair<String, String>, Long> pluginCounts = TransactionRunners.run(transactionRunner,
+                                                            (TxCallable<Map<ImmutablePair<String, String>, Long>>)
```

Comment:
I don't think this casting is necessary.

Reply:
Intellij complains when this cast is removed:
```
Ambiguous method call. Both
run(TransactionRunner, TxRunnable) in TransactionRunners
and
run(TransactionRunner, TxCallable<Map<ImmutablePair<String, String>, Long>>) in TransactionRunners
match
```